lib/spack/spack/build_systems/hip.py

```diff
@@ -98,6 +98,11 @@ class HipPackage(PackageBase):
     depends_on('hsa-rocr-dev', when='+hip')
     depends_on('hip', when='+hip')
 
+    # develop build of blt has fixes for hip, see blt pull
+    # requests from Nov 13 to 23 2020.
+    # This can be changed once blt release 0.3.7 is availible.
+    depends_on('blt@develop', type='build', when='+hip')
```

Comment:
This can change to `blt@0.3.7:`

Since `develop` is always higher than any other version, we can put in the constraint on the new release even before it's available.

Reply:
Didn't work, spack choked when I tried this. I'll keep it as develop for now

Reply:
What was the error message?

Reply:
It seemed to try to get that version directly, rather than realizing it is not there and reverting to develop:

==> Installing blt-0.3.7-6avnxgo56gkrfmvsidsqdpghwr2kcdy2
==> No binary for blt-0.3.7-6avnxgo56gkrfmvsidsqdpghwr2kcdy2 found: installing from source
==> Warning: There is no checksum on file to fetch blt@0.3.7 safely.
==>   Fetch anyway? [y/N] y
==> Fetching https://github.com/LLNL/blt/archive/v0.3.7.tar.gz
######################################################################## 100.0%
curl: (22) The requested URL returned error: 404 Not Found
==> Error: FetchError: All fetchers failed

/g/g17/taller1/CARE_SPACK/spack/lib/spack/spack/package.py:1331, in do_fetch:
       1328                                 self.spec.format('{name}{@version}'), ck_msg)
       1329
       1330        self.stage.create()
  >>   1331        err_msg = None if not self.manual_download else self.download_instr
       1332        self.stage.fetch(mirror_only, err_msg=err_msg)
       1333        self._fetch_time = time.time() - start_time
       1334

Reply:
Did you use `0.3.7` or `0.3.7:`?

Reply:
Ah! Good point. It seems to behave as you expected when I add the colon as 0.3.7:, 0.3.7 or newer